### PR TITLE
explorer: add log scale option to charts

### LINKED ISSUE #781: Chart log view
Need logarithmic scale view for
http://explorer.dcrdata.org/charts#pow-difficulty

### PR DESCRIPTION
**Pull request by buck54321**
Also adds a constant network target line to the ticket pool size chart.

Resolves #781 and #1054.

### CODE REVIEW

**Comment by chappjc** [comment]
Just testing and noticed this from eslint-loader:

```
/home/jon/github/decred/dcrdata/public/js/controllers/charts_controller.js
  269:49  error  Unnecessary use of boolean literals in conditional expression  no-unneeded-ternary
```

**Comment by chappjc** [comment]
Works great though.  Esp. nice hashrate, work, diff.

File: public/js/controllers/charts_controller.js [hunk 1]
```diff
@@ -254,7 +265,8 @@ export default class extends Controller {
     var gOptions = {
       rollPeriod: this.rollPeriod,
       zoomCallback: null,
-      drawCallback: null
+      drawCallback: null,
+      logscale: this.settings.scale === 'log' ? true : false
```
Comment (chappjc) [inline public/js/controllers/charts_controller.js]: This is where eslint complained during npm run build.

Comment (buck54321) [reply public/js/controllers/charts_controller.js]: Whoops. Thought I pushed this yesterday.

**Pull request merged**